### ISSUE TITLE
Mipmaps break after using LittleTiles doors

### ISSUE THREAD

**Issue by Sebastian11220** [CreativeMD/LittleTiles#150]
Hi. The title pretty much says it all, but here's some additional info:
The mipmaps flicker like they can't decide if they are enabled or not
Forge version: 14.23.5.2811
The only mods I have are:
-CreativeCore_v1.9.40_mc1.12.2
-LittleTiles_v1.5.0-pre127_mc1.12.2
so it can't be any other mods causing this
Some screenshots:
Before opening
![2019-03-14_18 54 34](https://user-images.githubusercontent.com/48565063/54380116-059bb600-468b-11e9-90d6-2d0951b85073.png)
After opening
![2019-03-14_18 55 00](https://user-images.githubusercontent.com/48565063/54380179-282dcf00-468b-11e9-852d-a99ca0163ed5.png)

**Comment by CreativeMD** [comment]
Does it happen with pre126?

**Comment by Sebastian11220** [comment]
It happened with pre125 so I assume it does happen with pre126

**Comment by CreativeMD** [comment]
Does it persists once the door is closed?

**Comment by Sebastian11220** [comment]
Yes it does and only goes away after full game restart. Also checked pre126 and so far no mipmap bugs

**Comment by CreativeMD** [comment]
Ok, so pre126 works just fine? That would be rather confusing.

**Comment by Sebastian11220** [comment]
Yep, but on pre126 anything I made transparent is fully opaque. Also it seems that it likes to crash a lot on my elevator door, even on pre127

**Comment by CreativeMD** [comment]
Can you send me the crash-reports?

**Comment by Sebastian11220** [comment]
Of course, this time with more mods loaded, but I doubt that they are the cause of the crashes
[crash-2019-03-14_19.53.57-client.txt](https://github.com/CreativeMD/LittleTiles/files/2968100/crash-2019-03-14_19.53.57-client.txt)
[crash-2019-03-14_19.58.07-client.txt](https://github.com/CreativeMD/LittleTiles/files/2968101/crash-2019-03-14_19.58.07-client.txt)

**Comment by CreativeMD** [comment]
Fixed your first issue. Can you send me your world so I can do some testing? Otherwise you can also explain when it happens and maybe provide some screenshots (or a video if possible)

**Comment by Sebastian11220** [comment]
Great to hear that got fixed. The world is 400mb and very private so I can't give it to you, but I managed to fix the crashes, one of the parts was "open" but it was supossed to be fixed, closing a door next to that part crashed the game, only noticed that when I tried to destroy that part and there was no hitbox so I used a wrench on it, hit destroy animation, the part stayed in place and I finally could destroy it. Clearly some invisible tiles magic thingy happened there, weird.
But that's not all, while flying around I noticed that open doors shift position when I move:
https://youtu.be/czEz8v-aEpQ
Also VanillaFix tells me that there's an error with LittleTiles almost every time I unload and load a chunk with an open door.

**Comment by CreativeMD** [comment]
Thanks, that's a big one. It's very small issue but makes it feel somewhat weird. Things look a lot more smooth now!

Can you send me the errors VanillaFix catches? Those are probably really important.

**Comment by Sebastian11220** [comment]
Ok, I have tested some things out and I think teleporting out of the chunk and back in creates this error.
The door then becomes invisible so I guess it's some unloading-loading error? Flying out of the chunk gives me the error and a slight lag spike, flying back in is the same.
Then I decided to do some more teleporting and it made every door that was unloaded with teleporting have a weird delay and they had no animations. Leaving the world and joining back crashed the game and trying to join again, crashes again, it only fixes itself after complete game restart.
Here are the crash reports:
[crash-2019-03-15_13.35.48-client.txt](https://github.com/CreativeMD/LittleTiles/files/2971207/crash-2019-03-15_13.35.48-client.txt) - Unloading and loading chunks with open doors
[crash-2019-03-15_13.49.43-client.txt](https://github.com/CreativeMD/LittleTiles/files/2971209/crash-2019-03-15_13.49.43-client.txt) - Crash when trying to join

**Comment by CreativeMD** [comment]
Ok, i will do some testing. If you want to you can try to it without VanillaFix and see if it still crashes.

**Comment by Sebastian11220** [comment]
Removed VanillaFix and just as I thought it didn't change anything. VanillaFix just allows me to keep playing after encoutering a crash, and it gives me a notification if there is a non crashing, but potentially very bad problem with a mod and then creates a crash report with that problem. So yeah, still crashes
[crash-2019-03-15_14.40.52-client.txt](https://github.com/CreativeMD/LittleTiles/files/2971339/crash-2019-03-15_14.40.52-client.txt)

**Comment by CreativeMD** [comment]
Things might not be fixed, but the crash reports should be more telling. So could you try to use this version and send the crash-reports (if they still persist)?
https://www.dropbox.com/s/63m1hwiogyhj0ob/LittleTiles_v1.5.0-pre129_mc1.12.2.jar?dl=1

**Comment by Sebastian11220** [comment]
Alright, got some bad news.
Firstly the mipmap bug is still happening.
Secondly the game crashes on anything that's more complex than just a door, so for example an iron base (fixed) and a wooden fan (advanced door) was enough to crash the game.
![2019-03-15_17 56 41](https://user-images.githubusercontent.com/48565063/54448648-8969a700-474c-11e9-99d0-12bcc3e9743f.png)
I tried to use my elevator door, crashed couple times and then it dissapeared completly, the rest of the elevator is fine, but I can't destroy it now because it says that the structure is not loaded. I'm now glad I made a backup of that world.
The unloading-loading bug seems to be gone now.
Here are the reports:
[crash-2019-03-15_18.09.31-client.txt](https://github.com/CreativeMD/LittleTiles/files/2972177/crash-2019-03-15_18.09.31-client.txt)
[crash-2019-03-15_17.57.28-client.txt](https://github.com/CreativeMD/LittleTiles/files/2972178/crash-2019-03-15_17.57.28-client.txt)
[crash-2019-03-15_17.57.26-server.txt](https://github.com/CreativeMD/LittleTiles/files/2972179/crash-2019-03-15_17.57.26-server.txt)
[crash-2019-03-15_17.56.21-client.txt](https://github.com/CreativeMD/LittleTiles/files/2972180/crash-2019-03-15_17.56.21-client.txt)
[crash-2019-03-15_17.38.05-client.txt](https://github.com/CreativeMD/LittleTiles/files/2972181/crash-2019-03-15_17.38.05-client.txt)
[crash-2019-03-15_17.31.54-server.txt](https://github.com/CreativeMD/LittleTiles/files/2972186/crash-2019-03-15_17.31.54-server.txt)

**Comment by CreativeMD** [comment]
I'm very confused by the mipmap bug reappearing. I tested it and everything worked fine. Is it still the exact some effect? Can you try to do another screenshot and see if it looks the same? Do you have any other mods installed such as Optifine?

Can you export the structure causing the issue. Just type in /lt-export put in the structure and post it on github. There seems to be an issue to connect the parent structures to the child structures.

**Comment by Sebastian11220** [comment]
Well I have tried it with just optifine installed, no other mods and no mipmap bug, so it's not optifine. I'll try to find the mod that's causing this.
Got a little confused with that export command, anyway here they are:
[Structure1.txt](https://github.com/CreativeMD/LittleTiles/files/2974005/Structure1.txt)
[Structure2.txt](https://github.com/CreativeMD/LittleTiles/files/2974006/Structure2.txt)
The second structure is as simple as it can get, the first one is my elevator.

**Comment by CreativeMD** [comment]
Can you try it without Optifine? Because it worked in vanilla.

**Comment by Sebastian11220** [comment]
Ahh, let me rephrase that. I tried it with mods: LittleTiles, CreativeCore and Optifine and mipmaps didn't get buggy, so it works with Optifine. That means other mod from my modpack is causing this.

**Comment by CreativeMD** [comment]
Oh right nice, I will investigate the structures you send me soon. Hopefully I can fix it.

**Comment by Sebastian11220** [comment]
I have found what's breaking the mipmaps now. It's the GUI with structure preview of Little Recipe and Advanced Recipe. Even tried it without Optifine to be sure, still breaks.
https://youtu.be/x2CFQvHd8Jo

**Comment by CreativeMD** [comment]
All your issues have been fixed. Uploaded pre130, which should be available soon. Thanks for your massive help! You provided some good material which made it easy for me to fix them. Hope everything works now :D

**Comment by Sebastian11220** [comment]
I'm glad I was able to help and thank you for fixing these issues :)
